eosdis-nasa/earthdata-pub-api
# Edpub 1333 form visibility

# Description

Adds backend ability to limit forms

## Spec


See Ticket: https://bugs.earthdata.nasa.gov/browse/EDPUB-1333

Related PR:
https://github.com/eosdis-nasa/earthdata-pub-dashboard/pull/80

---

## Validation

1. Make sure all merge request checks have passed (CI/CD).
2. Pull related branches locally.
3. Start application locally
4. Create separate accounts for DAAC_Staff and Data Producer Roles
5. Navigate to the swagger page and authorize with the DAAC_Staff account
6. In your db visualizer of choice set the daac_only column on one of the forms to true
7. On the swagger page under data in the  /api/data/form/{id} entry - enter the id for the form that you set to true in the previous step
8. Execute and verify you get a result
9. Under data in the /api/data/form execute the command without any input
10. Verify that all 3 forms return
11. Log into the Data Producer account and re-authorize swagger with the new bearer token
12. Repeat Step 7.
13. Verify that you get a "No results" error
14. Repeat Step 9.
15. Verify that only the forms that were not marked as daac_only return
